portfolio/src/main/webapp/script.js

```diff
@@ -61,3 +61,171 @@ function addRandomFunFact() {
 
 }
 
+/* 
+ * Creates the pagination bar given the number of pages and the previous active page number.
+ * Returns the current active page number.
+ */
+function createPagination(numPages, activePageNumber) {
+  // If the previous active page number is greater than the current number of pages, then set it to 1;
+  // Otherwise, use the previous as current active page number.
+  if (activePageNumber > numPages) {
+    activePageNumber = 1;
+  }
+  
+  pageListElement = document.getElementById('page-list');
+  pageListElement.innerHTML = '';
+
+  pageListElement.appendChild(createPageElement('Previous'));
+  for (var i = 1; i <= numPages; i++) {
+    if (i == activePageNumber) {
+      pageListElement.appendChild(createPageElement(i, true));
+    } else {
+      pageListElement.appendChild(createPageElement(i, false));
+    }
+  }
+  pageListElement.appendChild(createPageElement('Next'));
+
+  return activePageNumber;
+}
+
+/*
+ * Creates an <li> element for a pagination item whose a.innerText is 'text'.
+ * If 'active' is true, then add 'active' to its className.
+ */
+function createPageElement(text, active) {
+  const pageElement = document.createElement('li');
+  pageElement.className = 'page-item';
+  if (active) { pageElement.classList.add('active'); }
```

Comment:
To keep things clean and consistent, I would expand this.
```suggestion
  if (active) {
    pageElement.classList.add('active'); 
  }
```

Reply:
done